### PR TITLE
Custom hooks

### PR DESCRIPTION
**Pull request by arcoleburn**
adds exercise and solution for custom hooks bit

### CODE REVIEW

**Review by bmomberger-bitovi — APPROVED** [review]

File: custom-hooks/Exercise1/problem/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/problem/index.html]: This and the image can probably be removed.

File: custom-hooks/Exercise1/solution/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# React + TypeScript + Vite
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/solution/README.md:1]: Do you want to keep or remove this readme?

File: custom-hooks/Exercise1/solution/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "problem",
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/solution/package.json]: Maybe `@bitovi/trainings-custom-hooks` instead?

File: custom-hooks/Exercise1/solution/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+function App() {
+  const [on, toggle] = useToggle(true);
+
+  return (
+      <section>
+        <h1>UseToggle</h1>
+        <button disabled={on} onClick={() => toggle(true)}>
+          Turn On
+        </button>
+        <button disabled={!on} onClick={() => toggle(false)}>
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/solution/src/App.tsx:15]: I would [avoid disabling buttons](https://bitovi.slack.com/archives/C055KNS4ARL/p1697473839079379?thread_ts=1697473446.424119&cid=C055KNS4ARL) if you can.

File: custom-hooks/Exercise1/solution/src/Toggle.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    <div>
+      <label className="toggle">
+        <input
+          onChange={toggle}
+          className="toggle-checkbox"
+          type="checkbox"
+          checked={on}
+        />
+        <div className="toggle-switch"></div>
+        <span className="toggle-label">{on ? "On" : "Off"}</span>
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/solution/src/Toggle.tsx:20]: I haven’t seen this UI so I’m guessing here, please ignore if I’m wrong: is this changing the label based on the status of the checkbox? If so, I would keep this label stable (maybe On or Enabled) so there’s a consistent label for the checkbox.

File: custom-hooks/Exercise1/solution/src/index.css [hunk 6]
```diff
@@ -0,0 +1,69 @@
+:root {
+  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
+  line-height: 1.5;
+  font-weight: 400;
+
+  color-scheme: light dark;
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/solution/src/index.css:6]: I haven’t run your code so I don’t know this is the case, but I’m guessing that in dark mode you’ll have a flash of white and then it’ll switch to dark when first loading the page. If so, the only way I know to fix that is to include the `color-scheme` line in your HTML: https://github.com/bitovi/trainings/blob/024e679bafa2a7275b253997b3036ad0412e7185/incorporating-accessibility-into-your-dev-process/forms/solution/index.html#L8-L14

File: custom-hooks/Exercise1/solution/src/index.css [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  font-family: inherit;
+  background-color: #1a1a1a;
+  cursor: pointer;
+  transition: border-color 0.25s;
+}
+button:hover {
+  border-color: #646cff;
+}
+button:focus,
+button:focus-visible {
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/solution/src/index.css:54]: `:focus-visible` isn’t necessary if you have `:focus`, right?

File: custom-hooks/Exercise1/solution/src/vite-env.d.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+/// <reference types="vite/client" />
```
Comment (chasenlehara) [inline custom-hooks/Exercise1/solution/src/vite-env.d.ts:1]: Should this file be removed?

**Review by chasenlehara — APPROVED** [review]
I commented on a few nits here and there but otherwise looking good!

File: custom-hooks/Exercise1/solution/src/index.css (hunk 7 above)
Comment (arcoleburn) [reply custom-hooks/Exercise1/solution/src/index.css:54]: no clue, im a css luddite, ripped these from the ether. but it looks ok when it works

**Pull request merged**